daemon/container_unix.go

```diff
@@ -1054,29 +1140,76 @@ func (container *Container) releaseNetwork() {
 		return
 	}
 
-	n, err := container.daemon.netController.NetworkByID(nid)
-	if err != nil {
-		logrus.Errorf("error locating network id %s: %v", nid, err)
-		return
-	}
+	controller := container.daemon.netController
 
-	ep, err := n.EndpointByID(eid)
-	if err != nil {
-		logrus.Errorf("error locating endpoint id %s: %v", eid, err)
-		return
+	for _, ns := range networks {
+		n, err := findNetwork(controller, ns)
+		if err != nil {
+			continue
+		}
+		container.disconnectFromNetwork(n, false)
```

Comment:
disconnectFromNetwork is a common function called from Configuration event (such as `docker network disconnect`) which needs to absolutely remove configuration states. And also, it is called from an operational event (such as `docker stop <container>`), which should not update the network configuration settings. When the container starts again, it must be automatically connected to the networks it was connected earlier through user configuration.
That makes this path special and hence requires special handling.